lib/Conversion/StaticLogicToCalyx/StaticLogicToCalyx.cpp

```diff
@@ -429,60 +230,16 @@ class ComponentLoweringState {
   /// constitute the pipeline epilogue. Each inner vector consists of the groups
   /// for one stage.
   DenseMap<Operation *, SmallVector<SmallVector<StringAttr>>> pipelineEpilogue;
-
-  /// Block arg groups is a list of groups that should be sequentially
-  /// executed when passing control from the source to destination block.
-  /// Block arg groups are executed before blockScheduleables (akin to a
-  /// phi-node).
-  DenseMap<Block *, DenseMap<Block *, SmallVector<calyx::GroupOp>>>
-      blockArgGroups;
-
-  /// A while latch group is a group that should be sequentially executed when
-  /// finishing a while loop body. The execution of this group will write the
-  /// yield'ed loop body values to the iteration argument registers.
-  DenseMap<Operation *, calyx::GroupOp> whileLatchGroups;
-
-  /// A mapping from while ops to iteration argument registers.
-  DenseMap<Operation *, DenseMap<unsigned, calyx::RegisterOp>> whileIterRegs;
-
-  /// A mapping from memref's to their corresponding Calyx memory interface.
-  DenseMap<Value, calyx::MemoryInterface> memories;
-
-  /// A mapping between the source funcOp result indices and the corresponding
-  /// output port indices of this componentOp.
-  DenseMap<unsigned, unsigned> funcOpResultMapping;
 };
 
-/// ProgramLoweringState handles the current state of lowering of a Calyx
-/// program. It is mainly used as a key/value store for recording information
-/// during partial lowering, which is required at later lowering passes.
-class ProgramLoweringState {
+/// Handles the current state of lowering of a Calyx program. It is mainly used
+/// as a key/value store for recording information during partial lowering,
+/// which is required at later lowering passes.
+class ProgramLoweringState : public calyx::ProgramLoweringStateInterface {
```

Comment:
This feels a bit redundant with what is in SCFToCalyx. Does it make sense (for our current codebase) to not make `calyx::ProgramLoweringStateInterface` an interface, but instead just have 
```
template<typename TComponentLoweringState>
class ProgramLoweringState {
   ...
   DenseMap<Operation *, TComponentLoweringState> compStates;
}
```

Reply:
This initially wasn't possible, since ComponentLoweringState was returning a `ProgramLoweringState<ComponentLoweringState>`. However, I've removed this dependency; each use case was unnecessary.